packages/webview_flutter/webview_flutter_web/lib/src/http_request_factory.dart

```diff
@@ -55,27 +55,33 @@ class HttpRequestFactory {
   ///         // Do something with the response.
   ///     });
   ///
-  /// Note that requests for file:// URIs are only supported by Chrome extensions
+  /// Requests for file:// URIs are only supported by Chrome extensions
   /// with appropriate permissions in their manifest. Requests to file:// URIs
   /// will also never fail- the Future will always complete successfully, even
   /// when the file cannot be found.
   ///
   /// See also: [authorization headers](http://en.wikipedia.org/wiki/Basic_access_authentication).
-  Future<HttpRequest> request(String url,
-      {String? method,
-      bool? withCredentials,
-      String? responseType,
-      String? mimeType,
-      Map<String, String>? requestHeaders,
-      dynamic sendData,
-      void Function(ProgressEvent e)? onProgress}) {
-    return HttpRequest.request(url,
-        method: method,
-        withCredentials: withCredentials,
-        responseType: responseType,
-        mimeType: mimeType,
-        requestHeaders: requestHeaders,
-        sendData: sendData,
-        onProgress: onProgress);
+  Future<Object> request(
+    String url, {
+    String method = 'GET',
+    bool withCredentials = false,
+    String? mimeType,
+    Map<String, String>? requestHeaders,
+    Uint8List? sendData,
+  }) async {
+    final Map<String, String> headers = <String, String>{
+      if (mimeType != null) 'content-type': mimeType,
+      ...?requestHeaders,
+    };
+    return web.window
+        .fetch(
+            url.toJS,
+            web.RequestInit(
+              method: method,
+              body: sendData?.toJS,
+              credentials: withCredentials ? 'include' : 'same-origin',
+              headers: headers.jsify()! as web.HeadersInit,
```

Comment:
I don't think we have support for the JS Map type yet, so perhaps we should add a TODO for that?

In my opinion we can eventually get rid of the `jsify()` in the long run?

Not sure how @ditman feels about doing that, though?

Reply:
(Maybe something with `_createObjectLiteral`? [source](https://github.com/dart-lang/sdk/blob/60465149414572c8ca189d8f65fdb39795c4b97d/sdk/lib/js_interop/js_interop.dart#L107-L111))

Reply:
We could add a better constructor here if that's easier to read, but it's not going to do anything more complicated than just `jsify` :) e.g.

`JSObject.fromMap(Map m) => m.jsify() as JSObject;`

Reply:
@srujzs yeah, I'd like to have a `jsify` in non-nullable-object so I don't have to `Map.jsify()!` and maybe a `T jsifyAs<T>()` that does the casting for me, so I could replace:

```dart
headers.jsify()! as web.HeadersInit
```

by

```dart
headers.jsifyAs<web.HeadersInit>();
```

(Not that it saves too much, though)